test/generate_parse_fixture_yml.py

```diff
@@ -1,60 +1,142 @@
-                "a hash",
-                "# computed by SQLFluff when running the tests. Please run",
-                "# `python test/generate_parse_fixture_yml.py`  to generate them after "
-                "adding or",
-                "# altering SQL files.",
-                file=f,
-                sep="\n",
-            )
-            yaml.dump(r, f, default_flow_style=False, sort_keys=False)
-        else:
+        r: Optional[Dict[str, Optional[str]]] = None
+
+        if not tree:
             f.write("")
+            return
+
+        # Check we don't have any base types or unparsable sections
+        types = tree.type_set()
+        if "base" in types:
+            raise SQLParseError(f"Unnamed base section when parsing: {f.name}")
+        if "unparsable" in types:
+            raise SQLParseError(f"Could not parse: {f.name}")
+
+        records = tree.as_record(code_only=True, show_raw=True)
+        assert records, "TypeGuard"
+        r = dict([("_hash", _hash), *list(records.items())])
+        print(
+            "# YML test files are auto-generated from SQL files and should not be "
+            "edited by",
+            '# hand. To help enforce this, the "hash" field in the file must match '
+            "a hash",
+            "# computed by SQLFluff when running the tests. Please run",
+            "# `python test/generate_parse_fixture_yml.py`  to generate them after "
+            "adding or",
+            "# altering SQL files.",
+            file=f,
+            sep="\n",
+        )
+        yaml.dump(r, f, default_flow_style=False, sort_keys=False)
+        return
+
+
+def gather_file_list(
+    dialect: Optional[str] = None,
+    glob_match_pattern: Optional[str] = None,
+    new_only: bool = False,
+) -> List[_ParseExample]:
+    """Gather the list of files to generate fixtures for. Apply filters as required."""
+    parse_success_examples, _ = get_parse_fixtures()
+    if new_only:
+        parse_success_examples = [
+            example
+            for example in parse_success_examples
+            if not os.path.exists(_create_yaml_path(example))
+        ]
+
+    if dialect:
+        dialect = dialect.lower()
+        parse_success_examples = [
+            example for example in parse_success_examples if example[0] == dialect
+        ]
+        if len(parse_success_examples) == 0:
+            raise ValueError(f'Unknown Dialect "{dialect}"')
+
+    if not glob_match_pattern:
+        return parse_success_examples
+
+    regex = re.compile(fnmatch.translate(glob_match_pattern))
+    return [
+        example
+        for example in parse_success_examples
+        if regex.match(example[1]) is not None
+    ]
+
+
+@click.command()
+@click.option(
+    "--filter", "-f", default=None, help="A glob filter to apply to file names."
+)
+@click.option("--dialect", "-d", default=None, help="Filter to a given dialect.")
+@click.option(
+    "--new-only",
```

Comment:
It could also be cool to have a way to re-parse _modified_ files (where the `.sql` file is newer than the `.yml` file).

Reply:
Good idea, but can we push this feat for another PR?

Reply:
I went ahead and did it

Reply:
Amazing! 🙌

Reply:
Thanks, @OTooleMichael! About waiting, the answer was "yes", but it's too late now. 😄 Thanks again for the great work!!